### PR TITLE
Building Wintermute's Core Plugins (Issue #14)

### PR DESCRIPTION
**Pull request by jalcine**
This PR encompasses the work to be done for building out the core 
plugins of Wintermute in issue #14.

 + [x] wintermute-daemon (working on in #49).
 + [x] wintermute-heartbeat (pending)
 + ~~[ ] wintermute-transport-dbus (working on in #48).~~
 + [x] wintermute-transport-zeromq (done in #38).
 + ~~[ ] wintermute-transport-http (possibly punting on).~~
 + ~~[ ] wintermute-transport-gcm (possibly punting on).~~

### CODE REVIEW

**Comment by jalcine** [comment]
Sucks, CI is never going to work because I need at least CMake 2.8.12 and Travis is stuck on CMake 2.8.7. BOO! @github should support Drone.io.

File: src/transport-zeromq/receiver.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
```
Comment (jalcine) [inline src/transport-zeromq/receiver.cpp]: Going to pull out these changes and apply it on `develop` then rebase this in.
A cleaner header is nicer for folding.

File: src/transport-zeromq/receiver.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  socketPtr->setsockopt( ZMQ_SUBSCRIBE, subscriptionStr.c_str(), 1 );
+  wdebug("Subscribed to " + subscriptionStr + " via ZeroMQ.");
+}
+
+void ZMQReceiver::unsubscribeTo(const string& subscriptionStr)
+{
+}
+
+Message ZMQReceiver::receive()
+{
```
Comment (jalcine) [inline src/transport-zeromq/receiver.cpp]: Really need to explain what's happening in this function a bit more.

**Comment by bossjones** [comment]
There's no such thing as virtualenvs for CMake is there? @jalcine

**Comment by jalcine** [comment]
Nah, it's a hard dependency. CMake 2.8.7 is the most recent that Ubuntu 12.04 can support. I was experimenting with it in a Vagrant instance on the develop branch. :sob: Even from a point release tar-ball, it breaks on a later version.

It's fine though, I think I might be able to duplicate the functionality needed in a backwards compatible way. That of course, is for another day. For now, I have it working on 14.04 and will update the [platform list](https://github.com/jalcine/wintermute/blob/develop/INSTALL.markdown#platforms) when I get more of it working elsewhere.

Current tests on my laptop

Fast tests **FTW**!

**Comment by jalcine** [comment]
Nice and clean :+1:

**Comment by jalcine** [comment]
Sucks, my code is just not compatible with the version of Ubuntu used on Travis. The GCC compiler is at version 4.6.3 and I develop using 4.9 (VERY big jump, might as well have went to 5.0).

I'll just stick to using Vagrant to test other platforms.

**Comment by bossjones** [comment]
fuck thats so annoying ...

I was reading this thread re: making 14.04 ready for GCC 4.9.* etc ... seems as though they need some chef help getting there ... https://github.com/travis-ci/travis-ci/issues/2046

So maybe I can actually help you out indirectly by fixing up their chef shit :) lemme take a look and see where they're at.

**Comment by bossjones** [comment]
sent one of the travis ci dudes a tweet asking where I can help. I'm sure i'm going to need to know some of this C/CPP shit anyways if I want to install a better version of Gstreamer for scarlett from source haha. here: https://twitter.com/bossjones/status/553320240461783040

**Comment by jalcine** [comment]
Just peeped and subscribed to that conversation; hope it gets resolved soon.

Another option (since I've managed to get a 12.04 system going under Vagrant) is for me to properly have provisioning for Wintermute (in a development and production environment). Would be useful in the long run to have that.

**Comment by jalcine** [comment]
Dope, closed off the ZeroMQ plugin. That was good practice for handling other plugins down the route. Going to have to reevaluate some of these others plugins here.

**Comment by bossjones** [comment]
Off Wavy !!!!!!!!!!

Sent from my iPhone

On Jan 23, 2015, at 5:15 PM, Jacky Alcin? <notifications@github.com<mailto:notifications@github.com>> wrote:


Dope, closed off the ZeroMQ plugin. That was good practice for handling other plugins down the route. Going to have to reevaluate some of these others plugins here.

-

**Comment by jalcine** [comment]
Really got tired of how I can't get build results on Travis so I've added a build function for CMake 2.8.12. NO BARRIERS here!

**Comment by jalcine** [comment]
It's lit :fire: lol, [this output from `develop`](https://s3.amazonaws.com/archive.travis-ci.org/jobs/49115062/log.txt) shows a completed test, all passing!

**Comment by jalcine** [comment]
Another plugin built. This is getting easier as I go.

**Comment by jalcine** [comment]
Bam. Another plugin hammered out whilst improving the core library itself. This is _too_ easy now.

**Comment by jalcine** [comment]
Closed by release `0.0.4`

**Pull request closed without merging**